.github/workflows/build_cmake.yml

```diff
@@ -61,7 +61,26 @@ jobs:
       run: cmake --build ./build --config $BUILD_TYPE
 
     - name: Test
+      if: runner.os != 'Linux'
       working-directory: ./build
       # Execute tests defined by the CMake configuration.
       # See https://cmake.org/cmake/help/latest/manual/ctest.1.html for more detail
       run: ctest -C $BUILD_TYPE
+
+    - name: Test and coverage
+      if: runner.os == 'Linux'
+      working-directory: ./build
+      # Execute tests defined by the CMake configuration.
+      # See https://cmake.org/cmake/help/latest/manual/ctest.1.html for more detail
+      run: |
+          set -x
+          sudo apt install -y lcov gcovr
+          lcov --zerocounters --directory .
+          ctest -C $BUILD_TYPE
+          gcovr --root ../ --exclude '.*test/' --xml-pretty --xml coverage.xml
```

Comment:
the tests don't cover any lib/other files, so I should not exclude the folders ending with test
```suggestion
          gcovr --root ../ --xml-pretty --xml coverage.xml
```